presto-main/src/main/java/com/facebook/presto/sql/planner/plan/JoinNode.java

```diff
@@ -131,7 +131,6 @@ public JoinNode(
                 .addAll(right.getOutputVariables())
                 .build();
         checkArgument(new HashSet<>(inputVariables).containsAll(outputVariables), "Left and right join inputs do not contain all output variables");
-        checkArgument(!isCrossJoin() || inputVariables.size() == outputVariables.size(), "Cross join does not support output variables pruning or reordering");
```

Comment:
Why remove this line? If we still do not support output pruning for cross join, let's keep it.